converse.js

```diff
@@ -634,11 +635,35 @@
             },this), 200));
         };
 
+        this.ping = function (jid, timeout, success, error){
+            if (typeof jid === 'undefined'     || jid == null)     { jid = this.bare_jid; }
+                if (typeof timeout === 'undefined' || timeout == null) { timeout = 45; }
+                if (typeof success === 'undefined' ) { success = null; }
+                if (typeof error === 'undefined' ) { error = null; }
+                if (this.connection) this.connection.ping.ping( jid, success, error, timeout );
+        };
```

Comment:
If i add a test to detect if the server supports XEP-0199 Ping, it's the same problem, it will not work if Ping is not supported, but it would keep the connection anyway.
Do you prefer a proper way (with the test) or a non standart way ( but  keeping connections) ?